### DISCUSSION TITLE
why "{}".format(somevariable) is used in many places instead of f"{variable}" formatter?

### DISCUSSION THREAD

**Discussion by nikhil-swamix** [msiemens/tinydb#399] [General]
is it a coding style or has performance benefits? just curious as f"{variable}" is very intuitive than the format sting referred earlier.

**Comment by msiemens** [comment]
Well, the answer is relatively simple: f-strings were introduced in Python 3.6 which means that Python 3.5 was the most recent version that did _not_ support f-strings. Python 3.5 however was still supported up until September 2020 (see https://endoflife.date/python) and the code that uses `format` was written before that and hasn't been updated since 🙃 If you like, you can create a pull request that replaces the `format` calls with f-strings 🙂
